tests/suites/test_suite_psa_crypto.function

```diff
@@ -3152,6 +3152,862 @@ exit:
+        /* Pass data in parts */
+        part_data_size = PSA_AEAD_UPDATE_OUTPUT_SIZE( key_type, alg,
+                                                     ( size_t ) data_part_len );
+
+        ASSERT_ALLOC( part_data, part_data_size );
+
+        part_offset = 0;
+
+        while( part_offset <= input_data->len)
+        {
+            if( input_data->len - part_offset < ( uint32_t ) data_part_len )
+            {
+                part_length = input_data->len - part_offset;
+            }
+            else
+            {
+                part_length = data_part_len;
+            }
+
+            PSA_ASSERT( psa_aead_update( &operation,
+                                         ( input_data->x + part_offset ),
+                                         part_length, part_data,
+                                         part_data_size,
+                                         &output_part_length ) );
+
+            if( output_data && output_part_length )
+            {
+                memcpy( ( output_data + part_offset ), part_data,
+                        output_part_length );
+            }
+
+            part_offset += part_length;
+            output_length += output_part_length;
+        }
+    }
+    else
+    {
+        /* Pass whole data in one go */
+        PSA_ASSERT( psa_aead_update( &operation, input_data->x,
+                                         input_data->len, output_data,
+                                         output_size, &output_length ) );
+    }
+
+    PSA_ASSERT( psa_aead_finish( &operation, final_data,
+                                 PSA_AEAD_FINISH_OUTPUT_MAX_SIZE,
+                                 &output_part_length,
+                                 tag_buffer, tag_length,
+                                 &tag_size ) );
+
+    if( output_data && output_part_length )
+    {
+        memcpy( ( output_data + output_length ), final_data,
+                output_part_length );
+    }
+
+    TEST_EQUAL(tag_length, tag_size);
+
+    output_length += output_part_length;
+
+    if( output_data && tag_length )
+    {
+        memcpy( ( output_data + output_length ), tag_buffer, tag_length );
+    }
+
+    output_length += tag_length;
+
+    /* For all currently defined algorithms, PSA_AEAD_ENCRYPT_OUTPUT_SIZE
+     * should be exact. */
+    TEST_EQUAL( output_length,
+                PSA_AEAD_ENCRYPT_OUTPUT_SIZE( key_type, alg,
+                                              input_data->len ) );
+    TEST_ASSERT( output_length <=
+                 PSA_AEAD_ENCRYPT_OUTPUT_MAX_SIZE( input_data->len ) );
+
+    ASSERT_COMPARE( expected_result->x, expected_result->len,
+                    output_data, output_length );
+
+exit:
+    psa_destroy_key( key );
+    psa_aead_abort( &operation );
+    mbedtls_free( output_data );
+    mbedtls_free( part_data );
+    mbedtls_free( final_data );
+    PSA_DONE( );
+}
+/* END_CASE */
+
+/* BEGIN_CASE */
+void aead_multipart_encrypt_decrypt( int key_type_arg, data_t *key_data,
```

Comment:
What's the point of having encrypt-decrypt known-answer tests, in addition to known-answer tests for encrypt and for decrypt separately?

Reply:
I will admit I copied this from the one shot tests. The encrypt-decrypt tests do also use the output of encrypt to then decrypt which does add some extra coverage in my opinion, but I am guessing you want these removed?

Reply:
I don't see any point in having both. What added coverage is there? There are more tested values if there are more test cases with different data, sure, but why have multiple kinds of tests?

And yes, this is preexisting, but I don't see why this has to carry over to multipart tests.

By the way, apart from edge cases and bad-workflow tests, multipart operation could be tested solely by instrumenting the existing functions for one-shot operation and having them do split operation. It doesn't need separate test data.